types/src/access_path.rs

```diff
@@ -137,6 +139,7 @@ impl Accesses {
 
     /// Return the last access in the sequence
     pub fn last(&self) -> &Access {
+        assume!(self.0.last().is_some()); // follows from invariant
```

Comment:
that's not always true right? What if I create an access with `Accesses::empty()` and then end up calling this function?

Reply:
It is kind of a chicken-or-egg situation. Because of this usage, it would be an error to create an empty instance of Accesses, which is what the invariant in line 106 documents. It is also documented in the line below. Currently the code respects this requirement. Currently we can't enforce this, but at some point we could. Meanwhile it helps to warn developers via the comment.